src/pytest_workflow/plugin.py

```diff
@@ -172,26 +177,35 @@ def queue_workflow(self):
         Print statements are used to provide information to the user.
         This is shorter than using pytest's terminal reporter.
         """
-        # pylint: disable=protected-access
-        # Protected access needed to get the basetemp value.
-
-        basetemp = Path(str(self.config._tmp_path_factory.getbasetemp()))
-        self.tempdir = basetemp / Path(replace_whitespace(self.name, '_'))
+        basetemp = self.config.getoption("basetemp")
+        basetemp_path = (
+            Path(basetemp) if basetemp is not None
+            else Path(tempfile.mkdtemp(prefix="pytest_workflow_")))
+        self.tempdir = basetemp_path / Path(replace_whitespace(self.name, '_'))
+
+        # Remove the tempdir if it exists. This is needed for shutil.copytree
+        # to work properly.
+        if self.tempdir.exists():
```

Comment:
So this can only happen when the user supplies a `--basetemp` directory that already exists. Why not raise a big fat exception / error, and let the user select a non-existing directory rather than trying to clean it up ?

Reply:
Because if you type `pytest --help` you get this information:
```
test session debugging and configuration:
  --basetemp=dir        base temporary directory for this test run.(warning:
                        this directory is removed if it exists)
```

Reply:
Plus this allows you to run the same `pytest --basetemp <my_tmp>` command over and over and over. Which is convenient than having to set up a new tmp directory all the time.

Reply:
That reeks of bad design of pytest. Breaks principle of least surprise. But alas, if that's a pytest standard behavior, we kinda have to go with it.